core/services/keeper/registry_synchronizer_log_listener.go

```diff
@@ -14,7 +14,7 @@ func (rs *RegistrySynchronizer) JobID() int32 {
 func (rs *RegistrySynchronizer) HandleLog(broadcast log.Broadcast) {
 	eventLog := broadcast.DecodedLog()
 	if eventLog == nil || reflect.ValueOf(eventLog).IsNil() {
-		rs.logger.Errorf("HandleLog: ignoring nil value, type: %T", broadcast)
+		rs.logger.Panicf("HandleLog: ignoring nil value, type: %T", broadcast)
```

Comment:
not sure if this justifies Panic

Reply:
not trying to bike shed, but I guess for my own understanding... isn't "this should never happen" the same as `AssumptionViolation`, which is an Error level?

Reply:
I think the `HandleLog:` message prefix is redundant, no?
```suggestion
		rs.logger.Panicf("invariant violation: ignoring nil value, type: %T", broadcast)
```

Reply:
IMO this would be a programmer error and we should panic here, but as Jordan says it changes existing logic. This could also be AssumptionViolation